pymc3/distributions/dist_math.py

```diff
@@ -29,8 +29,16 @@ def bound(logp, *conditions):
 
 
 def alltrue(vals):
-    return tt.all([tt.all(1 * val) for val in vals])
+    """
+    Asserts truth of all elements in vals, across the lowest axis. This maintains
+    element-wise evaluations for multivariate inputs.
+    """
+    try:
+        return tt.all(tt.stack([1*val for val in vals]), axis=0)
+    except (TypeError, IndexError):
+        return tt.all([tt.all(1 * val) for val in vals])
```

Comment:
You run into situations where `vals` is not only of different types, but also of different lengths, so we need to deal with those potential combinations. `tt.stack` doesn't like heterogeneous dimensions. Haven't seen a case that tricks both yet, however.